modules/administration/pages/security-hardening.adoc

```diff
@@ -0,0 +1,54 @@
+[[security-hardening]]
+= Security Hardening
+
+While no implementation is ever completely secure from bad actors, there are some things you can do within {productname} to improve your security.
+
+This section contains a number of suggestions for hardening your security posture.
+The primary focus is to reduce the likelihood that a bad actor could access Salt clients in the case of a breach of the {productname} Server.
+In this case, we assume that the Salt master is running on the {productname} Server.
+
+To begin, there are some built-in mechanisms that make the {productname} Server less vulnerable to bad actors.
+The Salt master runs as a non-privileged user, and the Salt API only listens on localhost, not an external address.
+
+Additionally, Salt clients cannot send jobs to the Salt master in the default configuration.
+This can be manually adjusted, but it is not recommended.
+
+There are a limited number of ports required for clients to communicate with the server, which simplifies network management.
+The only required ports are 4505 and 4506.
+Ports 80 and 443 are only used for package repositories, and they secured with SSL trusted certificates and GPG keys.
+
+The server never initiates a TCP handshake with a client.
+Clients initiate every connection, and the communication is bi-drectional after initiation.
+
+Each client creates a set of RSA keys.
+The public key is accepted by the server when a new system is added.
+If the keys are secure on the server, only clients with the correct private keys can be authenticated.
+The keys should be secured by specific users.
+
+For extra key security, you can implement a key policy.
+A key policy that allows only pre-seeded keys on the master is the most secure.
+Additionally, you can implement a key policy that allows clients to communicate only with a master that has a specific digital fingerprint.
+
+You can also implement a key size limit to increase security.
+
+All data between the server and clients is encrypted with a shared AES key.
+The key is is regenerated every 24 hours, or whenever a client is removed.
+
+Client pillar data is also encrypted with AES.
+In this case, a new AES session key is created every time a pillar is refreshed.
+
+
+You do not need to run Salt commands as root.
+This requires some additional configuration, so that administrators can run Salt commands using their user ID.
+Adjusting this also allows for improved auditing.
+
+You can add additional user-level controls, such as only allowing specific Salt commands.
+When you have selected a subset of allowed Salt commands, you can further limit the arguments that are accepted for each command.
```

Comment:
How to configure this?

Reply:
As per previous, I think this would be done with states, so the relevant xref is to https://opensource.suse.com/doc-susemanager/suse-manager/salt/salt-states.html